src/lib.rs

```diff
@@ -124,6 +131,38 @@ macro_rules! bitfield_impl {
             $self.0[i] $op $rhs.0[i];
         }
     }};
+    (new for struct $name:ident([$t:ty]); $($rest:tt)*) => {
+        impl<T: AsMut<[$t]> + Default> $name<T> {
+            bitfield_constructor!{T::default(); () -> {}; $($rest)*}
+        }
+    };
+    (new for struct $name:ident($t:ty); $($rest:tt)*) => {
+        impl $name {
+            bitfield_constructor!{0; () -> {}; $($rest)*}
+        }
+    };
+    (new{$new:ident ($($setter_name:ident: $setter_type:ty),*$(,)?)} for struct $name:ident([$t:ty]); $($rest:tt)*) => {
+        impl<T: AsMut<[$t]> + Default> $name<T> {
+            pub fn $new($($setter_name: $setter_type),*) -> Self {
+                let mut value = Self(T::default());
+                $(
+                    value.$setter_name($setter_name);
+                )*
+                value
+            }
+        }
+    };
+    (new{$new:ident ($($setter_name:ident: $setter_type:ty),*$(,)?)} for struct $name:ident($t:ty); $($rest:tt)*) => {
+        impl $name {
+            pub fn $new($($setter_name: $setter_type),*) -> Self {
+                let mut value = Self(0);
```

Comment:
Is there any reason to use 0 directly instead of `default` here?

Reply:
I thought it would make for *slightly* more readable code. Since this version should only be used for integer inner-types, `0` will always work. I don't see anything wrong with using `default`.